CONTRIBUTING.md

```diff
@@ -3,9 +3,200 @@
 Metal3 projects are [Apache 2.0 licensed](LICENSE) and accept contributions via
 GitHub pull requests.
 
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Certificate of Origin](#certificate-of-origin)
+- [Finding Things That Need Help](#finding-things-that-need-help)
+- [Versioning](#versioning)
+  - [Codebase](#codebase)
+    - [Backporting](#backporting)
+- [Branches](#branches)
+- [Contributing a Patch](#contributing-a-patch)
+- [Backporting a Patch](#backporting-a-patch)
+- [Breaking Changes](#breaking-changes)
+  - [Merge Approval](#merge-approval)
+  - [Google Doc Viewing Permissions](#google-doc-viewing-permissions)
+  - [Issue and Pull Request Management](#issue-and-pull-request-management)
+  - [Commands and Workflow](#commands-and-workflow)
+- [Release Process](#release-process)
+  - [Exact Steps](#exact-steps)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
 ## Certificate of Origin
 
 By contributing to this project you agree to the Developer Certificate of
 Origin (DCO). This document was created by the Linux Kernel community and is a
 simple statement that you, as a contributor, have the legal right to make the
 contribution. See the [DCO](DCO) file for details.
+
+## Finding Things That Need Help
+
+If you're new to the project and want to help, but don't know where to start, we
+have a semi-curated list of issues that
+should not need deep knowledge of the system. [Have a look and see if anything
+sounds interesting](https://github.com/metal3-io/baremetal-operator/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+Alternatively, read some of the docs on other controllers and try to write your
+own, file and fix any/all issues that come up, including gaps in documentation!
+
+## Versioning
+
+### Codebase
+
+Baremetal Operator doesn't follow release cadence and versioning of upstream Kubernetes semantic versioning. This codebase guarantees the following:
+
+- A (*minor*) release CAN include:
+  - Introduction of new API versions, or new Kinds.
+  - Compatible API changes like field additions, deprecation notices, etc.
+  - Breaking API changes for deprecated APIs, fields, or code.
+  - Features, promotion or removal of feature gates.
+  - And more!
+
+- A (*patch*) release SHOULD only include backwards compatible set of bugfixes.
+
+These guarantees extend to all code exposed in our Go Module, including
+*types from dependencies in public APIs*.
+Types and functions not in public APIs are not considered part of the guarantee.
+
+#### Backporting
+
+We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
+backport MUST not be breaking for either API or behavioral changes.
+We generally do not accept PRs against older release branches.
```

Comment:
since for now we only have a "main" branch, this might confuse a reader. Either we have to remove it or make it clear that once we start having release branches in the future, backporting is the option we would have.

Reply:
++